lib/id3c/cli/command/manifest.py

```diff
@@ -487,10 +476,85 @@ def select_columns(table: pandas.DataFrame, name: str) -> pandas.DataFrame:
     return table[matches]
 
 
-def qc_barcodes(df: pandas.DataFrame, columns: Iterable) -> pandas.DataFrame:
+def perform_qc(sample_column: str, collection_column: str, barcode_columns: Set[str],
+    parsed_manifest: pandas.DataFrame) -> pandas.DataFrame:
+    """
+    Perform quality control on the manifest data, dropping rows which violate
+    our standards for complete and accurate data.
+    """
+    parsed_manifest = drop_missing_barcodes(sample_column, collection_column, parsed_manifest)
+
+    parsed_manifest = drop_missing_racks(parsed_manifest)
+
+    # Drop any rows that have duplicated barcodes
+    parsed_manifest = deduplicate_barcodes(parsed_manifest, barcode_columns)
```

Comment:
Should drop racks and drop barcode ordering be switched here? If we drop one of the duplicate barcode pair in the `drop_missing_racks` function then we will ingest the other one of the duplicate pair. Does that make sense?

Reply:
Ah yes, thank you! That was another question I had but forgot to list at the top of this PR.

I wasn't sure which ordering was best. I took this initial pass thinking maybe the duplicate barcode with an incomplete entry (missing racks) is OK to drop and fine to ingest the other matching barcode with a complete entry. 

But it seems like you disagree. Which one seems preferable? I can see that dropping duplicate barcodes before doing more filtering would be safer.

Reply:
I like to keep the duplicate dropping to the very last step because previous steps might get rid of some of those anyway.

If we have one sample record that has no rack info and another record with the same identifiers and with rack info I think we would want to use that second record.

Reply:
If the second record is completed at a later date, they are still true duplicates and both should be dropped.

Reply:
We'd get that record on the next run, so I'm OK dropping it on the first run. I'm OK putting the duplication dropper above the rack dropper.

Reply:
Is it worth evaluating date to try to pull in additional records? Now that you lay it out like that, @joverlee521 , I think conditionally dropping barcodes first based on date, filtering where racks is not None, and then dropping barcodes again feels a little clunky just in the off chance we might rescue a couple of rows here and there. I'd rather put the onus on the lab to clean their manifest data instead of using obscure logic to ingest most of it is by far the more robust solution. The lab would also appreciate knowing if they have data entry errors (which they'd learn if we don't ingest certain barcodes at all).

Reply:
Ah I just saw Chris's comment. Looks like we all agree now.

Reply:
I pushed a new commit that deduplicates first.